pandas/tests/test_duplicate_labels.py

```diff
@@ -0,0 +1,398 @@
+"""Tests dealing with the NDFrame.allows_duplicates."""
```

Comment:
i would put this under pandas/tests/indexing

Reply:
Feels more closely realted to ndframe to me, than indexing. Perhaps `pandas/tests/generic`?